plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergTableWithCustomLocation.java

```diff
@@ -0,0 +1,178 @@
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SelectedRole;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static io.trino.spi.security.SelectedRole.Type.ROLE;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestIcebergTableWithCustomLocation
+        extends AbstractTestQueryFramework
+{
+    private FileHiveMetastore metastore;
+
+    private static final String CATALOG = "iceberg";
+    private static final String SCHEMA = "iceberg_schema";
+    private static final String TABLE = "iceberg_table";
+    private static final String TABLE_RENAMED = "iceberg_table_renamed";
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setIdentity(Identity.forUser("hive")
+                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
+                        .build())
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
+
+        File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data").toFile();
+
+        HdfsConfig hdfsConfig = new HdfsConfig();
+        HdfsConfiguration hdfsConfiguration = new HiveHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), ImmutableSet.of());
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hdfsConfig, new NoHdfsAuthentication());
+
+        metastore = new FileHiveMetastore(
+                new NodeVersion("test_version"),
+                hdfsEnvironment,
+                new MetastoreConfig(),
+                new FileHiveMetastoreConfig()
+                        .setCatalogDirectory(baseDir.toURI().toString())
+                        .setMetastoreUser("test"));
+
+        TestingIcebergPlugin icebergPlugin = new TestingIcebergPlugin(metastore, false);
+
+        queryRunner.installPlugin(icebergPlugin);
+        queryRunner.createCatalog(CATALOG, "iceberg", ImmutableMap.of("iceberg.unique-table-location", "true"));
+        queryRunner.installPlugin(new TestingHivePlugin(metastore));
+        queryRunner.createCatalog("hive", "hive", ImmutableMap.of("hive.security", "sql-standard"));
+        return queryRunner;
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        assertQuerySucceeds(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
```

Comment:
We can use `iceberg.default` schema, so that we don't have to setup things.

actually, the `QueryRunner` should have iceberg catalog, and the chosen schema set as session's default, so that you don't need to qualify tables in all the queries

Reply:
done